nncf/quantization/algorithms/pipeline.py

```diff
@@ -113,10 +113,13 @@ def run_step(
 
         pipeline_steps = self._remove_unsupported_algorithms(get_backend(model))
         pipeline_step = pipeline_steps[step_index]
+
         for algorithm in pipeline_step[:-1]:
             current_model = algorithm.apply(current_model, current_graph, step_statistics)
             current_graph = NNCFGraphFactory.create(current_model)
+
         current_model = pipeline_step[-1].apply(current_model, current_graph, step_statistics)
+        step_statistics.remove_statistic_points(pipeline_step[-1]._algorithm_key)
```

Comment:
Also, instead of using _private_ `_algorithm_key`, you can extend the existing `Algorithm` class with the same property.
We'll very much appreciate this contribution.

Reply:
# Just to confirm:

Do you expect it to look something like this?

```python
class Algorithm:
...

def __init__(self, algorithm_name):
    self.algorithm_key = algorithm_name

...
```

If we do it this way, won't we have to initialize this property everywhere **Algorithms** are used?

# Another idea
I was also thinking to have encapsulation like this:

```python
class Algorithm:
...

def get_algorithm_name(self):
    return self._algorithm_key

...
```

Reply:
The **algorithm_name** is not the same thing as the **algorithm_key**.
Please refer to the example: https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algorithms/min_max/algorithm.py#L225